chemist-aws/src/main/scala/AwsDiscovery.scala

```diff
@@ -157,54 +166,51 @@ class AwsDiscovery(
    * @see funnel.chemist.AwsDiscovery.lookupOne
    */
   protected def lookupMany(ids: Seq[String]): Task[Seq[AwsInstance]] = {
-    def lookInCache: (Seq[String],Seq[AwsInstance]) =
-      ids.map(id => id -> cache.get(id)
-        ).foldLeft[(Seq[String],Seq[AwsInstance])]((Seq.empty, Seq.empty)){ (a,b) =>
+    def lookInCache: (Seq[String], Seq[AwsInstance]) =
+      ids.map(
+        id => id -> cache.get(id)
+      ).foldLeft[(Seq[String], Seq[AwsInstance])]((Seq.empty, Seq.empty)) {
+        (a,b) =>
           val (ids,instances) = a
           b match {
             case (id,Some(instance)) => (ids, instances :+ instance)
             case (id,None) => (ids :+ id, instances)
           }
-        }
+      }
 
     def lookInAws(specificIds: Seq[String]): Task[Seq[AwsInstance]] =
-      LookupManyAws.timeTask {
-        for {
-          a <- EC2.reservations(specificIds)(ec2)
-          _  = log.debug(s"lookupMany, reservations = ${a.length}")
-          b <- Task.now(a.flatMap(_.getInstances.asScala.map(fromAWSInstance)))
-          (fails,successes) = b.toVector.separate
-          _  = log.info(s"lookupMany, failed to validate ${fails.length} instances.")
-          _  = log.debug(s"lookupMany, validated ${successes.length} instances.")
-          _  = fails.foreach(x => log.error(s"lookupMany, failed to validate '$x'"))
-        } yield successes
-      }
+      if (specificIds.isEmpty)
```

Comment:
imho, empty seq is valid input => i do not expect lookup to fail on empty input.
However, `EC2.reservations` will choke on empty seq. 

Before this was special cased in the callee code, i am making this function to be safe for all inputs (and this allows to simplify code where it is called).